src/pages/registration/contributors_tab/components/EditAffiliationModal.tsx

```diff
@@ -0,0 +1,118 @@
+  affiliationModalIsOpen,
+  toggleAffiliationModal,
+  affiliationToEdit,
+  authorName,
+  affiliations = [],
+  baseFieldName,
+}: EditAffiliationModalProps) => {
+  const { t } = useTranslation();
+  const { setFieldValue } = useFormikContext<Registration>();
+  const dispatch = useDispatch();
+  const institution = findAncestor(affiliationToEdit);
+  const institutionQuery = useFetchOrganization(institution.id);
+
+  const editAffiliation = (newAffiliationId: string) => {
+    if (!newAffiliationId) {
+      return;
+    }
+
+    const affiliationToChangeIndex = affiliations.findIndex(
+      (a) => a.type === 'Organization' && a.id === affiliationToEdit.id
+    );
+
+    if (affiliationToChangeIndex < 0) {
+      return;
+    }
+
+    // If user tries to change it into already existing affiliation
+    if (
+      affiliations.some((affiliation) => affiliation.type === 'Organization' && affiliation.id === newAffiliationId)
+    ) {
+      dispatch(setNotification({ message: t('registration.contributors.add_duplicate_affiliation'), variant: 'info' }));
+      return;
+    }
+
+    const newAffiliation: Affiliation = {
+      type: 'Organization',
+      id: newAffiliationId,
+    };
+
+    const updatedAffiliations = [...affiliations];
+
+    // Replace old affiliation
+    updatedAffiliations[affiliationToChangeIndex] = newAffiliation;
+
+    setFieldValue(`${baseFieldName}.${SpecificContributorFieldNames.Affiliations}`, updatedAffiliations);
+    toggleAffiliationModal();
+  };
+
+  return (
+    <Modal
+      open={affiliationModalIsOpen}
+      onClose={() => {
+        toggleAffiliationModal();
+      }}
+      maxWidth="md"
+      fullWidth={true}
+      headingText={t('registration.contributors.edit_affiliation')}
+      dataTestId="edit-affiliation-modal">
+      <Trans i18nKey="registration.contributors.edit_affiliation_helper_text" components={[<Typography paragraph />]} />
+      {authorName && (
+        <Box
+          sx={{
+            bgcolor: 'secondary.main',
+            borderRadius: '0.25rem',
+            padding: '0.5rem 0.75rem',
+            marginBottom: '2rem',
+          }}>
+          <Typography>
+            {t('common.contributor')}: <b>{authorName}</b>
+          </Typography>
+        </Box>
+      )}
+      {institutionQuery.isPending ? (
+        <SelectInstitutionSkeleton />
+      ) : institutionQuery.data ? (
+        <SelectInstitutionForm
+          saveAffiliation={editAffiliation}
+          onCancel={toggleAffiliationModal}
+          initialValues={{
+            unit: institutionQuery.data,
+            subunit:
+              affiliationToEdit.id !== institutionQuery.data.id
+                ? findDescendantWithId(institutionQuery.data, affiliationToEdit.id)
+                : null,
+            selectedSuggestedAffiliationId: '',
+          }}
+        />
+      ) : (
+        <Typography sx={{ fontStyle: 'italic' }}>[{t('feedback.error.get_organization')}]</Typography>
```

Comment:
If we fail to fetch institution we will display this error message:

<img width="941" alt="image" src="https://github.com/BIBSYSDEV/NVA-Frontend/assets/1130244/e14c3ad9-5e5e-4bdd-92be-0a1a12e70fa1">